### PR TITLE
Add android platform KDD

### LINKED ISSUE #916: KDD for android platform

### PR DESCRIPTION
**Pull request by andreievg**
closes #916 

This is a draft KDD, likely needs more input from the rest of the team

### CODE REVIEW

**Review by mark-prins — COMMENTED** [review]
Thanks Andrei, nicely written up!
It might be helpful to clarify what bare bones means, so that we know we are talking about the same thing. 

I don't think that capacitor involves much dev setup, though more than nothing. 

I have a few questions that is like to explore before we commit either way

- can the app bar be removed from the bare bones app? (I assume yes, but was unable to compile and check)
- can all of the capacitor plugins in use be ported? (Printing, barcode scanner, keyboard) would be good to test
- can we try to replicate the capacitor functionality which tells the fe that a plugin is available?

Leaning towards option 2, as #3 isn't available yet, and the current implementation of #1 includes the fe bundle twice, as well as requiring a package or two. Concerned that we are adding to the Dev overhead with a custom implementation, but think it's ok, it's not code that changes much

**Comment by andreievg** [comment]
@mark-prins thanks for review, I am going to rewrite this one now. It seems that i was wrong about:

> It looks like we can only serve a bundled app (although it can be served from url, url needs to be know at compile time), this affects requirement number `6`

As for the questions, I tried porting one plugin, and even though it's possible, it's not trivial. After some further digging around I realised we can serve bundle from any server, anyways I'll update KDD now

File: decisions/android-framework.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+No frameworks available for this yet, possibly Tauri (issue from september 2022 says they have mobile alpha in next few weeks).
+
+## Extra
+
+Requirement `2` hasn't been mentioned in both options, since I think it's independent of the options, we can point both options to use bundles served on dev computer for hot reload and debugging and both options use Android Studio for native code debugging.
+
+## Decision
+
+Original suggestion was to go with `Option 2`, but after some research we realised that being able to serve bundle from arbitary server was possible with Capacitor (although it's a bi hacky), and trying to port a plugin to Native barebone application was quite difficult. Would now suggest `Option 1`, the cons can be metigated by better explanation of what Capacitor is and how we use it. Also the `cons` of `Option 2` can lead to quite a lot of extra work
```
Comment (mark-prins) [inline decisions/android-framework.md:62]: soz. just some pedantry.

```suggestion
Original suggestion was to go with `Option 2`, but after some research we realised that being able to serve bundle from arbitrary server was possible with Capacitor (although it's a bi hacky), and trying to port a plugin to Native barebone application was quite difficult. Would now suggest `Option 1`, the cons can be mitigated by better explanation of what Capacitor is and how we use it. Also the `cons` of `Option 2` can lead to quite a lot of extra work
```

File: decisions/android-framework.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Lastly woudl suggest keeping an eye on Option 3 and re-writting this KDD when Option 3 becomes feasible (tauri mobile available and stable)
```
Comment (mark-prins) [inline decisions/android-framework.md:64]: :nit
```suggestion
Lastly would suggest keeping an eye on Option 3 and re-writing this KDD when Option 3 becomes feasible (tauri mobile available and stable)
```

**Review by mark-prins — APPROVED** [review]
🙏

**Comment by andreievg** [comment]
🙏 for review and suggestions @mark-prins

**Pull request merged**